### PR TITLE
Update JShintrb to support ES6 syntax

### PR DESCRIPTION
**Pull request by gylaz**
* Enables the usage of `"esnext": true` config option

https://trello.com/c/tUPpmibV

### CODE REVIEW

**Comment by salbertson** [comment]
Yay, looks good to me. :+1:

File: Gemfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ gem "font-awesome-rails"
 gem "haml-rails"
 gem "high_voltage"
 gem "jquery-rails"
-gem "jshintrb"
+gem "jshintrb", github: "thoughtbot/jshintrb"
```
Comment (teoljungberg) [inline Gemfile]: Maybe add the `HEAD` commit?

**Comment by croaky** [comment]
Awesome. This seems Intercom message-worthy, and updates to the homepage and config docs?

**Comment by salbertson** [comment]
@croaky :+1:

File: Gemfile (hunk 1 above)
Comment (gylaz) [inline Gemfile]: It's locked/specified in `Gemfile.lock`. What benefit do you see by using that here?

**Comment by gylaz** [comment]
We might want to sit on this a bit before sending out messages. One issue that I can think of is that there are a few different file extensions people use with `ES6` syntax, like `.es6`, `.es` (besides the `.js`). @croaky @salbertson What do you think about giving this a week to bake before a public message?

**Comment by croaky** [comment]
@gylaz Sounds good regarding holding off on public message.

**Pull request merged**